os/fs/smartfs/smartfs_smart.c

```diff
@@ -2127,6 +2164,39 @@ int smartfs_rename(struct inode *mountpt, const char *oldrelpath, const char *ne
 }
 
 /****************************************************************************
+ * Name: smartfs_stat_common
+ *
+ * Description:
+ *   Return information about a directory entry
+ *
+ ****************************************************************************/
+
+static void smartfs_stat_common(FAR struct smartfs_mountpt_s *fs,
+                                FAR struct smartfs_entry_s *entry,
+                                FAR struct stat *buf)
+{
+	/* Initialize the stat structure */
+
```

Comment:
Shouldn't we validate buf here?  In case smartfs_stat_common() is called from other api's with NULL buf..
Similarly, fs and entry pointers..

Reply:
Moved the checks to callsites (parent functions).